### PR TITLE
Add unit tests for user management

### PR DESCRIPTION
**Pull request by joe-singleton-itdev**
Add unit tests for creating users, removing users and changing user passwords through the parser

### CODE REVIEW

**Comment by itdevreviewboard** [comment]
Can one of the MCVirt admins verify this patch?

File: source/usr/lib/mcvirt/test/auth_tests.py [hunk 1]
```diff
@@ -219,3 +219,45 @@ def test_delete_non_existant_superuser(self):
...
+        )
+
+        # Try to create an RPC connection with the old credentials and check an AuthenticationError
+        # is raised
+        with self.assertRaises(AuthenticationError):
+            new_rpc = Connection(self.TEST_USERNAME, self.TEST_PASSWORD)
+
+        # Create a new RPC connection with the new credentials
+        try:
+            new_rpc = Connection(self.TEST_USERNAME, new_password)
```
Comment (MatthewJohn) [inline source/usr/lib/mcvirt/test/auth_tests.py]: You could do an assertion here to ensure that this connection can do something, e.g. do a list or something?

File: source/usr/lib/mcvirt/test/auth_tests.py [hunk 2]
```diff
@@ -219,3 +219,45 @@ def test_delete_non_existant_superuser(self):
...
+        self.rpc.annotate_object(new_user)
+        new_user.delete()
+
+    def test_remove_user_account(self):
+        """Delete a user through the parser"""
+        self.user_to_delete = self.create_test_user('user-to-delete', 'pass')
+        self.parser.parse_arguments('user remove user-to-delete')
+        # Check that the user no longer exists
+        self.assertFalse('user-to-delete' in MCVirtConfig().get_config()['users'])
+        self.user_to_delete = None
```
Comment (MatthewJohn) [inline source/usr/lib/mcvirt/test/auth_tests.py]: Is it worth trying the password generation?

File: source/usr/lib/mcvirt/test/auth_tests.py [hunk 3]
```diff
@@ -219,3 +219,45 @@ def test_delete_non_existant_superuser(self):
...
+    def test_add_new_user(self):
+        """Create a new user through the parser"""
+        username = 'brand-new-user'
+        password = 'pass'
+        self.parser.parse_arguments('user create %s --user-password %s' %
+            (username, password)
+        )
+
+        # Check the new user exists
+        self.assertTrue(username in MCVirtConfig().get_config()['users'])
```
Comment (MatthewJohn) [inline source/usr/lib/mcvirt/test/auth_tests.py]: I would use:self.user_factory.get_user_by_username

File: source/usr/lib/mcvirt/test/auth_tests.py [hunk 4]
```diff
@@ -219,3 +219,45 @@ def test_delete_non_existant_superuser(self):
...
+        new_user = self.user_factory.get_user_by_username(username)
+        self.rpc.annotate_object(new_user)
+        new_user.delete()
+
+    def test_remove_user_account(self):
+        """Delete a user through the parser"""
+        self.user_to_delete = self.create_test_user('user-to-delete', 'pass')
+        self.parser.parse_arguments('user remove user-to-delete')
+        # Check that the user no longer exists
+        self.assertFalse('user-to-delete' in MCVirtConfig().get_config()['users'])
```
Comment (MatthewJohn) [inline source/usr/lib/mcvirt/test/auth_tests.py]: I would use:

    with self.assertRaises(UserDoesNotExistException):
        self.user_factory.get_user_by_username('user-to-delete')

File: source/usr/lib/mcvirt/test/auth_tests.py [hunk 5]
```diff
@@ -219,3 +219,45 @@ def test_delete_non_existant_superuser(self):
...
+        self.assertTrue(username in MCVirtConfig().get_config()['users'])
+
+        # Delete the new user
+        new_user = self.user_factory.get_user_by_username(username)
+        self.rpc.annotate_object(new_user)
+        new_user.delete()
+
+    def test_remove_user_account(self):
+        """Delete a user through the parser"""
+        self.user_to_delete = self.create_test_user('user-to-delete', 'pass')
```
Comment (MatthewJohn) [inline source/usr/lib/mcvirt/test/auth_tests.py]: I would put 'user-to-delete' in a variable :) Or possibly put a define a second test username in setUp and use it for 'user-to-delete' and 'brand-new-user'.

Then in the tearDown/setup, delete this user if it exists :)

File: source/usr/lib/mcvirt/test/unit_test_bootstrap.py [hunk 6]
```diff
@@ -52,12 +52,12 @@ def __init__(self):
         online_migrate_test_suite = OnlineMigrateTests.suite()
         OnlineMigrateTests.RPC_DAEMON = self.daemon
         self.all_tests = unittest.TestSuite([
-            virtual_machine_test_suite,
-            network_test_suite,
-            drbd_test_suite,
-            update_test_suite,
-            node_test_suite,
-            online_migrate_test_suite,
+            # virtual_machine_test_suite,
```
Comment (MatthewJohn) [inline source/usr/lib/mcvirt/test/unit_test_bootstrap.py]: Please uncomment these!

**Comment by MatthewJohn** [comment]
ok to test

**Pull request merged**